sql/core/src/main/scala/org/apache/spark/sql/execution/limit.scala

```diff
@@ -283,20 +283,7 @@ case class TakeOrderedAndProjectExec(
   }
 
   override def executeCollect(): Array[InternalRow] = {
-    val orderingSatisfies = SortOrder.orderingSatisfies(child.outputOrdering, sortOrder)
```

Comment:
In this PR, we are adding more case to benefit from TakeOrderedAndProjectExec.
is there any reason for us to modify executeCollect?

Reply:
Originally to prevent inconsistent order of query results. please see https://github.com/apache/spark/pull/37118 for more details.